File: stvs/src/test/java/edu/kit/iti/formal/stvs/logic/io/xml/XmlSessionImporterMalformedXmlTest.java
package edu.kit.iti.formal.stvs.logic.io.xml;

import edu.kit.iti.formal.stvs.logic.io.ExportException;
import edu.kit.iti.formal.stvs.logic.io.ImportException;
import edu.kit.iti.formal.stvs.logic.io.ImporterFacade;
import edu.kit.iti.formal.stvs.model.config.*;
import edu.kit.iti.formal.stvs.model.config.History;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.io.File;
import java.io.IOException;
import java.net.URISyntaxException;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * @author Benjamin Alt
 */
@RunWith(Parameterized.class)
public class XmlSessionImporterMalformedXmlTest {
  private File file;

  @Parameterized.Parameters
  public static Collection<File> testFiles() throws URISyntaxException {
    List<File> testFiles = new ArrayList<>();
    for (int i = 1; i < 6; i++) {
      testFiles.add(new File(XmlConstraintSpecImporter.class.getResource(
          "session_invalid_xml_" + i + ".xml").toURI().getPath()));
    }
    return testFiles;
  }

  public XmlSessionImporterMalformedXmlTest(File input) {
    this.file = input;
  }

  /**
   * Test whether importing malformed XML throws ImportExceptions.
   */
  @Test(expected=ImportException.class)
  public void sessionImporterMalformedXmlTest() throws URISyntaxException, ImportException,
      IOException, ExportException {
      ImporterFacade.importSession(file, ImporterFacade.ImportFormat.XML, new GlobalConfig(), new
          History());
  }
}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/logic/io/xml/XmlSessionImporterMalformedXmlTest.kt
package edu.kit.iti.formal.stvs.logic.io.xml

import edu.kit.iti.formal.stvs.logic.io.ExportException
import edu.kit.iti.formal.stvs.logic.io.ImportException
import edu.kit.iti.formal.stvs.logic.io.ImporterFacade
import edu.kit.iti.formal.stvs.model.config.GlobalConfig
import edu.kit.iti.formal.stvs.model.config.History
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.io.IOException
import java.io.InputStream
import java.net.URISyntaxException
import kotlin.test.assertFailsWith

/**
 * @author Benjamin Alt
 */
class XmlSessionImporterMalformedXmlTest {
    /**
     * Test whether importing malformed XML throws ImportExceptions.
     */
    @ParameterizedTest//(expected = ImportException::class)
    @MethodSource("testFiles")
    @Throws(URISyntaxException::class, ImportException::class, IOException::class, ExportException::class)
    fun sessionImporterMalformedXmlTest(name: String, file: InputStream) {
        assertFailsWith<ImportException> { ImporterFacade.importSession(file, GlobalConfig(), History()) }
    }

    companion object {
        @JvmStatic
        @Throws(URISyntaxException::class)
        fun testFiles() =
            (1..5).map { i ->
                Arguments.of(
                    "session_invalid_xml_$i.xml",
                    XmlConstraintSpecImporter::class.java.getResourceAsStream("session_invalid_xml_$i.xml")
                )
            }
    }
}
